src/parser.ts

```diff
@@ -188,25 +179,26 @@ const parseRouteEndpoint = async (route: IRoute, basePath: string, config: IGene
 
 	const methods = await Promise.all(
 		map(routeMethods, async (method) => {
-			// get permission middlewares and workflow of endpoint
-			const permissionHandlerPromises: Promise<{ groupName?: string } & ILocation>[] = []
+			// get middlewares and workflow of endpoint
+			const middlewaresHandlerPromises: Promise<{ closure: string; middlewareName: string } & ILocation>[] = []
 			let workflowHandlerPromise: Promise<ILocation> | null = null
 			let security: ISecurity[] = []
 
 			forEach(route.stack, (handler) => {
-				forEach(config.permissions, (permission) => {
-					if (handler.name === permission.middlewareName) {
-						permissionHandlerPromises.push(
+				forEach(config.middlewares, (middleware) => {
+					if (handler.name === middleware.middlewareName) {
+						middlewaresHandlerPromises.push(
 							// eslint-disable-next-line no-async-promise-executor
 							new Promise(async (resolve) => {
+								const closure = middleware.closure === 'default' ? 'exports.default' : middleware.closure
 								const location = await locate(handler.handle, {
-									closure: permission.closure === 'default' ? 'exports.default' : permission.closure,
-									paramName: permission.paramName,
-									parser: permission.parser
+									closure: middleware.closure === 'default' ? 'exports.default' : middleware.closure,
+									maxParamDepth: middleware.maxParamDepth
 								})
 
 								resolve({
-									groupName: permission.groupName ?? undefined,
+									middlewareName: handler.name,
+									closure,
```

Comment:
@kot-1999 Why do you need closure is result of this promise? I do not see place where it is used

Reply:
@daniel-zavacky forgot to remove